File: archive/c/c-sharp/LongestWord.cs
if (args is not [var sentence] || string.IsNullOrWhiteSpace(sentence))
    return Usage();

int max = 0,
    cur = 0;

foreach (char c in sentence)
{
    if (char.IsWhiteSpace(c))
    {
        cur = 0;
        continue;
    }

    cur++;
    max = Math.Max(cur, max);
}

Console.WriteLine(max);
return 0;

static int Usage()
{
    Console.Error.WriteLine("Usage: please provide a string");
    return 1;
}

File: archive/f/f-sharp/LongestWord.fs
open System

module LongestWord =
    let run (s: string) =
        s
        |> Seq.fold
            (fun (maxLen, currLen) ch ->
                if Char.IsWhiteSpace ch then
                    (max maxLen currLen, 0)
                else
                    (maxLen, currLen + 1))
            (0, 0)
        |> fun (maxLen, currLen) -> max maxLen currLen

module Helpers =
    let usage = "Usage: please provide a string"

    let parseArgs argv =
        match argv with
        | [| s |] when not (String.IsNullOrEmpty s) -> Ok s
        | _ -> Error usage

    let handleResult =
        function
        | Ok result ->
            printfn "%d" result
            0
        | Error msg ->
            eprintfn "%s" msg
            1

[<EntryPoint>]
let main argv =
    argv |> Helpers.parseArgs |> Result.map LongestWord.run |> Helpers.handleResult
